x-pack/test/security_solution_api_integration/test_suites/entity_analytics/risk_engine/trial_license_complete_tier/risk_engine_so_config.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { riskEngineConfigurationTypeName } from '@kbn/security-solution-plugin/server/lib/entity_analytics/risk_engine/saved_object';
+import { FtrProviderContext } from '../../../../ftr_provider_context';
+import {
+  riskEngineRouteHelpersFactory,
+  getRiskEngineConfigSO,
+  waitForRiskEngineRun,
+  waitForRiskEngineTaskToBeGone,
+} from '../../utils';
+
+export default ({ getService }: FtrProviderContext) => {
+  const spaceName = 'space1';
+  const supertest = getService('supertest');
+  const riskEngineRoutes = riskEngineRouteHelpersFactory(supertest);
+  const riskEngineRoutesForNamespace = riskEngineRouteHelpersFactory(supertest, spaceName);
+  const kibanaServer = getService('kibanaServer');
+  const esArchiver = getService('esArchiver');
+
+  describe('@ess @ serverless @serverless QA risk_engine_so_update_config', () => {
+    before(async () => {
+      const soId = await kibanaServer.savedObjects.find({
+        type: riskEngineConfigurationTypeName,
+        space: spaceName,
+      });
+      if (soId.saved_objects.length !== 0) {
+        await kibanaServer.savedObjects.delete({
+          type: riskEngineConfigurationTypeName,
+          space: spaceName,
+          id: soId.saved_objects[0].id,
+        });
+      }
+      const soId2 = await kibanaServer.savedObjects.find({
+        type: riskEngineConfigurationTypeName,
+      });
+      if (soId2.saved_objects.length !== 0) {
+        await kibanaServer.savedObjects.delete({
+          type: riskEngineConfigurationTypeName,
+          id: soId2.saved_objects[0].id,
+        });
+      }
+      await esArchiver.load('x-pack/test/functional/es_archives/security_solution/ecs_compliant');
+    });
+
+    after(async () => {
+      const soId = await kibanaServer.savedObjects.find({
+        type: riskEngineConfigurationTypeName,
+        space: spaceName,
+      });
+      if (soId.saved_objects.length !== 0) {
+        await kibanaServer.savedObjects.delete({
+          type: riskEngineConfigurationTypeName,
+          space: spaceName,
+          id: soId.saved_objects[0].id,
+        });
+      }
+      const soId2 = await kibanaServer.savedObjects.find({
+        type: riskEngineConfigurationTypeName,
+      });
+      if (soId2.saved_objects.length !== 0) {
+        await kibanaServer.savedObjects.delete({
+          type: riskEngineConfigurationTypeName,
+          id: soId2.saved_objects[0].id,
+        });
+      }
+      await esArchiver.unload('x-pack/test/functional/es_archives/security_solution/ecs_compliant');
+    });
+
+    it('should include the right keys as per the update', async () => {
+      await riskEngineRoutes.init();
+      await waitForRiskEngineRun;
```

Comment:
This doesn't wait because you are calling the function
.
```suggestion
    await waitForRiskEngineRun({ log, supertest });
```